rbc/irtools.py

```diff
@@ -295,3 +355,20 @@ def compile_IR(ir):
     engine.run_static_constructors()
 
     return engine
+
+
+@extending.intrinsic
+def printf(typingctx, format_type, arg0=nb_types.void,
+           arg1=nb_types.void, arg2=nb_types.void, arg3=nb_types.void,
+           arg4=nb_types.void, arg5=nb_types.void, arg6=nb_types.void,
+           arg7=nb_types.void, arg8=nb_types.void, arg9=nb_types.void):
+    """printf that can be called from Numba jit-decorated functions.
+    """
+    if isinstance(format_type, nb_types.StringLiteral):
+        sig = nb_types.void(format_type, arg0, arg1, arg2, arg3, arg4, arg5,
+                            arg6, arg7, arg8, arg9)
+
+        def codegen(context, builder, signature, args):
+            cgutils.printf(builder, format_type.literal_value, *args[1:])
+
+        return sig, codegen
```

Comment:
You can create a variadic intrinsic in Numba using `types.BaseTuple`
```suggestion
@extending.intrinsic
def printf(typingctx, format_type, *args):
    if isinstance(format_type, types.StringLiteral):
        sig = types.void(format_type, types.BaseTuple.from_types(args))

        def codegen(context, builder, signature, args):
            cgutils.printf(builder, format_type.literal_value, *args[1:]) 

        return sig, codegen
```

Reply:
We could send this to numba

Reply:
Thanks for the variadic hint.
Sent:
https://github.com/numba/numba/issues/5679